shared/actions/fs/index.js

```diff
@@ -710,15 +710,31 @@ const moveOrCopy = (state, action) => {
       kbfs: Constants.fsPathToRpcPathString(
         Types.pathConcat(
           action.payload.destinationParentPath,
-          Types.getPathName(state.fs.moveOrCopy.sourceItemPath)
+          state.fs.destinationPicker.source.type === 'move-or-copy'
+            ? Types.getPathName(state.fs.destinationPicker.source.path)
+            : state.fs.destinationPicker.source.type === 'incoming-share'
+              ? Types.getLocalPathName(state.fs.destinationPicker.source.localPath)
+              : ''
```

Comment:
Perhaps there can be a guard at beginning

```
if (state.fs.destinationPicker.source.type === 'none') {
return
}
```

so we don't need to use an empty string here.

Reply:
I prefer this, since we need to return something.

Reply:
It's a Saga handler. We can return `null` meaning we'd do nothing for that action, which makes more sense than calling a RPC as we don't have a source path.